server.go

```diff
@@ -936,3 +756,280 @@ func (s *Server) handlerHistory(w http.ResponseWriter, r *http.Request) {
 	}
 }
+
+func (s *Server) handleJikan(data *dataAddMovie, w http.ResponseWriter, r *http.Request, sourcelink string) ([]string, error) {
+
+	jikanEnabled, err := s.data.GetCfgBool("JikanEnabled", DefaultJikanEnabled)
+	if err != nil {
+		s.l.Error("Error while retriving config value 'JikanEnabled':\n %v", err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return nil, err
+	}
+
+	s.l.Debug("jikanEnabled: %v", jikanEnabled)
+
+	if !jikanEnabled {
+		s.l.Debug("Aborting Jikan autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Jikan API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Jikan not enabled")
+	} else {
+		// Get Data from MAL (jikan api)
+		rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/anime\/([0-9]*)`)
+		match := rgx.FindStringSubmatch(sourcelink)
+		var id string
+		if len(match) < 2 {
+			s.l.Debug("Regex match didn't find the anime id in %v", sourcelink)
+			data.ErrorMessage = append(data.ErrorMessage, "Could not retrive anime id from provided link, did you input a manga link?")
+			data.ErrLinks = true
+			return nil, fmt.Errorf("Could not retrive anime id from link")
+		} else {
+			id = match[1]
+		}
+
+		bannedTypesString, err := s.data.GetCfgString(ConfigJikanBannedTypes, DefaultJikanBannedTypes)
+
+		if err != nil {
+			s.l.Error("Error while retriving config value 'JikanBannedTypes':\n %v", err)
+			s.doError(
+				http.StatusInternalServerError,
+				"something went wrong :C",
+				w, r)
+			return nil, err
+		}
+
+		bannedTypes := strings.Split(bannedTypesString, ",")
+
+		maxEpisodes, err := s.data.GetCfgInt(ConfigJikanMaxEpisodes, DefaultJikanMaxEpisodes)
+
+		if err != nil {
+			s.l.Error("Error while retriving config value 'JikanMaxEpisodes':\n %v", err)
+			s.doError(
+				http.StatusInternalServerError,
+				"something went wrong :C",
+				w, r)
+			return nil, err
+		}
+
+		sourceAPI := jikan{id: id, l: s.l, excludedTypes: bannedTypes, maxEpisodes: maxEpisodes}
+
+		// Request data from API
+		results, err := getMovieData(&sourceAPI)
+
+		if err != nil {
+			s.l.Error("Error while accessing Jikan API: %v", err)
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
+			return nil, err
+		}
+
+		return results, nil
+	}
+}
+
+func (s *Server) handleTmdb(data *dataAddMovie, w http.ResponseWriter, r *http.Request, sourcelink string) ([]string, error) {
+
+	tmdbEnabled, err := s.data.GetCfgBool("TmdbEnabled", DefaultTmdbEnabled)
+	if err != nil {
+		s.l.Error("Error while retriving config value 'TmdbEnabled':\n %v", err)
+		data.ErrorMessage = append(data.ErrorMessage, "Something went wrong :C")
+		return nil, err
+	}
+
+	if !tmdbEnabled {
+		s.l.Debug("Aborting Tmdb autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Tmdb API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Tmdb not enabled")
+	} else {
```

Comment:
The `else` should be removed here and just have everything under the `if` block.  The `if` block returns from the function rendering the `else` a bit redundant.

Reply:
While i kinda see your point there i would like to ask you to rethink that request.
The whole purpose of this PR is to increase readability and reduce the time for someone else to work themself into the codebase.
In my opinion this "if {} else {}" structure improves readability by a lot opposed to the "if {return} // now the error case will never get here" approach.

Reply:
I would actually argue that the `if { return }` without the `else` block is *more* readable.  It's returning because of an error and is similar to the `if err != nil { return }` pattern that is already used everywhere.

Additionally, this is pattern is also more idiomatic.  See here: https://golang.org/doc/effective_go.html#if

Reply:
Alright, i see your point - i'll rework it since i actually dont really mind the change anyways, but its nice to talk about it :).

Reply:
Thanks!

Reply:
Fixed in 407f9c8